.circleci/config.yml

```diff
@@ -124,6 +124,131 @@ jobs:
+              echo "No .toml files changed in superchain/configs, skipping codegen check"
+              exit 0
+            fi
+
+            # Extract chain_ids from changed .toml files
+            CHAIN_IDS=""
+            CHAIN_ID_COUNT=0
+            CHANGED_FILE_COUNT=0
+
+            for file in $CHANGED_FILES; do
+              CHANGED_FILE_COUNT=$((CHANGED_FILE_COUNT + 1))
+              # Use yq to extract chain_id from TOML
+              CHAIN_ID=$(yq -p=toml -o=json '.chain_id' "$file" | grep -v "null" | tr -d '"')
+
+              if [ -n "$CHAIN_ID" ]; then
+                if [ -z "$CHAIN_IDS" ]; then
+                  CHAIN_IDS="$CHAIN_ID"
+                else
+                  CHAIN_IDS="$CHAIN_IDS,$CHAIN_ID"
+                fi
+                CHAIN_ID_COUNT=$((CHAIN_ID_COUNT + 1))
+                echo "Found chain_id $CHAIN_ID in $file"
+              fi
+            done
+            echo "Found $CHAIN_ID_COUNT chain_ids in $CHANGED_FILE_COUNT files"
+
+            cd ops
+            if [ "$CHAIN_ID_COUNT" -ne "$CHANGED_FILE_COUNT" ] && [ "$CHANGED_FILE_COUNT" -gt 0 ]; then
+              # This accounts for changes to superchain.toml files
+              echo "Running codegen for all chains\n"
+              go run ./cmd/codegen \
+                --l1-rpc-urls="<< pipeline.parameters.sepolia_rpc_url >>,<< pipeline.parameters.mainnet_rpc_url >>"
+            else
+              echo "Running codegen for the following chain_ids: $CHAIN_IDS\n"
+              go run ./cmd/codegen \
+                --l1-rpc-urls="<< pipeline.parameters.sepolia_rpc_url >>,<< pipeline.parameters.mainnet_rpc_url >>" \
+                --chain-ids="$CHAIN_IDS"
+            fi
+
+            if [ -n "$(git status --porcelain)" ] ; then
+              echo "\n❌ Changes detected after running codegen. Run the following command locally and commit the changes:\n"
+              echo "go run ./cmd/codegen \\"
+              # Show the appropriate command flags based on which mode was run
+              if [ "$CHAIN_ID_COUNT" -ne "$CHANGED_FILE_COUNT" ] && [ "$CHANGED_FILE_COUNT" -gt 0 ]; then
+                echo "  --l1-rpc-urls=\"<urls>\""
+              else
+                echo "  --l1-rpc-urls=\"<urls>\" \\"
+                echo "  --chain-ids=\"$CHAIN_IDS\""
+              fi
+              exit 1
+            else
+              echo "\n✅ All codegen files are up to date"
+            fi
+
+  codegen-sync-all:
+    circleci_ip_ranges: true
+    docker:
+      - image: <<pipeline.parameters.default_docker_image>>
+    steps:
+      - utils/checkout-with-mise
+      - run:
+          name: Execute codegen
+          working_directory: ops
+          command: |
+            go run ./cmd/codegen/main.go --l1-rpc-urls="<< pipeline.parameters.sepolia_rpc_url >>,<< pipeline.parameters.mainnet_rpc_url >>"
+      - run:
+          name: Create PR if changes detected
+          working_directory: .
+          command: |
```

Comment:
I have not tested this yet. Might be best to extract this daily-ci job from this pr and put into a follow-up pr

Reply:
Removed in https://github.com/ethereum-optimism/superchain-registry/pull/951/commits/dbf4ebda3c1c7bd4bb963baa5f05670285c80a04 to make this pr cleaner/easier to review